File: sdk/cpp/src/api/asset.cpp
// Copyright 2026 Hypermesh Foundation. All rights reserved.
// Licensed under the Business Source License 1.1.

#include <hypermesh/api/asset.hpp>

namespace hypermesh {

AssetList AssetApi::list() const {
    auto j = http_.get("/api/v1/asset/list");
    return j.get<AssetList>();
}

} // namespace hypermesh

File: sdk/typescript/src/api/asset.ts
import type { HttpClient } from "../client.js";
import type { AssetListResponse } from "../types.js";

export class AssetApi {
  constructor(private readonly http: HttpClient) {}

  async list(): Promise<AssetListResponse> {
    return this.http.get<AssetListResponse>("/api/v1/asset/list");
  }
}
